### ISSUE TITLE
Combining sampleplot and plot

### ISSUE THREAD

**Issue by willtebbutt** [JuliaGaussianProcesses/AbstractGPs.jl#220]
It's great that we have these plotting recipes separated for situations in which you only want one or the other, but quite often you really _do_ want both together. I wonder whether it wouldn't make sense to add a third method that just does both at the same time?

**Comment by theogf** [comment]
I would disagree, I think it's kinda nice to be able to split your plotting via
```julia
plot(gp)
sampleplot!(gp)
```
and have more control. But that's just my opinion

**Comment by willtebbutt** [comment]
Agreed, which is why I'm not proposing to get rid of those two methods, I'm just suggesting we might add a third recipe so that I can write something like
```julia
plot_everything(gp)
```
and it gives me all of the things.

**Comment by devmotion** [comment]
I think the main pain point here would be to distinguish between the keyword arguments for `plot` and `sampleplot`. I guess the only clean way would be to define `plot_everything` and `plot_everything!` without any `@recipe` magic and use named tuples for the kwargs of the two functions, e.g.,
```julia
function  plot_everything(args...; plotkwargs=(), sampleplotkwargs=())
    plt = plot(args...; plotkwargs...)
    sampleplot!(plt, args...; sampleplotkwargs...)
end
function  plot_everything!(args...; plotkwargs=(), sampleplotkwargs=())
    plt = plot!(args...; plotkwargs...)
    sampleplot!(plt, args...; sampleplotkwargs...)
end
```
and

**Comment by willtebbutt** [comment]
Hmmmm... that's a good point. It seems like it would get quite messy quite quickly. I'm going to drop this and not bring it up for a third time!
